gemrb/GUIScripts/GUICommon.py

```diff
@@ -502,35 +505,36 @@ def IsDualClassed(actor, verbose):
 	DualedFrom = GemRB.GetPlayerStat (actor, IE_MC_FLAGS) & MC_WAS_ANY_CLASS
 
 	if verbose:
-		DualInfo = []
 		KitIndex = GetKitIndex (actor)
+		if KitIndex:
+			KittedClass = CommonTables.KitList.GetValue (KitIndex, 7)
+			KittedClassIndex = CommonTables.Classes.FindValue ("ID", KittedClass)
+		else:
+			KittedClassIndex = 0
 
 		if DualedFrom > 0: # first (previous) class of the dual class
 			FirstClassIndex = CommonTables.Classes.FindValue ("MC_WAS_ID", DualedFrom)
-			if KitIndex:
-				DualInfo.append (1)
-				DualInfo.append (KitIndex)
-			else:
-				DualInfo.append (2)
-				DualInfo.append (FirstClassIndex)
 
 			# use the first class of the multiclass bunch that isn't the same as the first class
-			Mask = 1
 			for i in range (1,16):
+				Mask = 1 << (i - 1)
 				if Multi & Mask:
 					ClassIndex = CommonTables.Classes.FindValue ("ID", i)
 					if ClassIndex == FirstClassIndex:
-						Mask = 1 << i
 						continue
-					DualInfo.append (ClassIndex)
+					SecondClassIndex = ClassIndex
 					break
-				Mask = 1 << i
-			if len(DualInfo) != 3:
+			else:
```

Comment:
nice, not a construct one sees often!

Reply:
Yes, things like that are what makes Python such a beautiful language.